app_flutter/lib/build_dashboard_page.dart

```diff
@@ -130,6 +131,89 @@ class BuildDashboardPageState extends State<BuildDashboardPage> {
     );
   }
 
+  List<PopupMenuEntry<String>> _getTaskKey(bool isDark) {
+    const List<String> taskStatuses = <String>[
+      TaskBox.statusFailed,
+      TaskBox.statusNew,
+      TaskBox.statusSkipped,
+      TaskBox.statusSucceeded,
+      TaskBox.statusInProgress,
+    ];
+    final List<PopupMenuEntry<String>> key = <PopupMenuEntry<String>>[];
+
+    for (final String status in taskStatuses) {
+      key.add(PopupMenuItem<String>(
+        value: status,
+        child: Wrap(
+          crossAxisAlignment: WrapCrossAlignment.center,
+          children: <Widget>[
+            const SizedBox(width: 10.0),
+            SizedBox.fromSize(
+                size: const Size.square(TaskBox.cellSize), child: Container(color: TaskBox.statusColor[status])),
+            const SizedBox(width: 10.0),
+            Text(status),
+          ],
+        ),
+      ));
+      key.add(const PopupMenuDivider());
+    }
+
+    key.add(PopupMenuItem<String>(
```

Comment:
It would be easier to read if this was a function that took in a SizedBox and text description to show flaky tasks, reruns, and status colors use the same layout.